tree/treeplayer/src/TSimpleAnalysis.cxx

```diff
@@ -0,0 +1,384 @@
+// @(#)root/treeplayer:$Id$
+// Author: Luca Giommi   22/08/16
+
+/*************************************************************************
+ * Copyright (C) 1995-2016, Rene Brun and Fons Rademakers.               *
+ * All rights reserved.                                                  *
+ *                                                                       *
+ * For the licensing terms see $ROOTSYS/LICENSE.                         *
+ * For the list of contributors see $ROOTSYS/README/CREDITS.             *
+ *************************************************************************/
+
+#include "TSimpleAnalysis.h"
+
+#include "TFile.h"
+#include "TChain.h"
+#include "TH1.h"
+#include "TError.h"
+#include "TKey.h"
+
+#include <string>
+#include <fstream>
+#include <vector>
+#include <map>
+#include <iostream>
+
+/** \class TSimpleAnalysis
+A TSimpleAnalysis object allow to, given an input file, create a
+.root file in which are saved histograms that the user wants to
```

Comment:
A TSimpleAnalysis object creates histograms from a TChain. These histograms are stored to an output file. The histogrammed expressions, their cuts, the input and output files are configured through a simple config file that allows comments starting with '#'.